### PR TITLE
Blog - How To Simplify Kubernetes Configuration Using Custom Resources And Controllers

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
@jalletto

File: blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+
+Now, apply all the controller configurations to your [cluster](/blog/kube-bench,
+
+~~~{.bash caption=">_"}
+cd  ..  &&  kubectl apply -f controller
+~~~
+
+You should see the following confirmation:
+
+~~~{.bash caption=">_"}
```
Comment (jalletto) [inline blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md:442]:
```suggestion
~~~{.bash caption="output"}
```

File: blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+deployment.apps/apptemplate-controller created
+~~~
+
+Now that your custom controller is running, the `sample-service` microservice you created earlier using the apptemplate custom resource should have all its resources (namespace, deployment, services, ) deployed. To confirm this, check for the microservice namespace
+
+~~~{.bash caption=">_"}
+kubectl get namespaces
+~~~
+
+You should be able to find the ${environment}-sample-service namespace, where the `environment` is the value of the environment in the apptemplate e.g development
```
Comment (jalletto) [inline blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md:455]:
```suggestion
You should be able to find the `${environment}-sample-service namespace`, where the `environment` is the value of the environment in the apptemplate e.g development
```

File: blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+
+~~~{.bash caption=">_"}
```
Comment (jalletto) [inline blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md:457]:
```suggestion
~~~{.bash caption="output"}
```

File: blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md [hunk 4]
```diff
@@ -0,0 +1,513 @@
...
+kubectl get pods -n development-sample-service
+~~~
+
+~~~{.bash caption=">_"}
+
+Sample-service-deployment-69d89bf574-2j4xf      1/1      Running   0    56s
+sample-service-deployment-69d89bf574-m5gnx      1/1     Running    0    56s
+~~~
```
Comment (jalletto) [inline blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md:473]:
```suggestion

~~~{.bash caption=">_"}
kubectl get pods -n development-sample-service
~~~

-

~~~{.bash caption=">_"}

Sample-service-deployment-69d89bf574-2j4xf      1/1      Running   0    56s
sample-service-deployment-69d89bf574-m5gnx      1/1     Running    0    56s
~~~
```

File: blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md [hunk 5]
```diff
@@ -0,0 +1,513 @@
...
+
+~~~{.bash caption=">_"}
+kubectl get apptemplate
+~~~
+
+~~~{.bash caption=">_"}
+NAME                  AGE
+sample-service        13m
+sample-service-2      1m
+~~~
```
Comment (jalletto) [inline blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md:507]:
```suggestion

~~~{.bash caption=">_"}
kubectl get apptemplate
~~~

-

~~~{.bash caption="output"}
NAME                  AGE
sample-service        13m
sample-service-2      1m
~~~
```

File: blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md [hunk 6]
```diff
@@ -0,0 +1,513 @@
...
+
+> The configuration files in the repository above are mainly for example purposes. The aim is to show you how you can make use of the microservice-specific configuration created from the custom resource to modify the main configuration template.
+
+This configuration template contains deployment.yaml, service.yaml, namespace.yaml files and an `extras` folder. Each of your configuration files has access to all the specifications defined in the appTemplate custom resource. For instance, check out the deployment.yaml code
+
+~~~{.bash caption=">_"}
+cat deployment.yaml
+~~~
+
+~~~{.yaml caption="deployment.yaml"}
```
Comment (jalletto) [inline blog/_posts/2022-11-06-simplify-k8s-using-cr-cp.md:250]:
```suggestion

~~~{.bash caption=">_"}
cat deployment.yaml
~~~

-

~~~{.yaml caption="deployment.yaml"}
```

**Review by jalletto — CHANGES_REQUESTED** [review]

**Review by jalletto — APPROVED** [review]

**Pull request merged**